python/pyiceberg/serializers.py

```diff
@@ -54,7 +54,8 @@ def table_metadata(input_file: InputFile, encoding: str = "utf-8") -> TableMetad
             TableMetadata: A table metadata instance
 
         """
-        return FromByteStream.table_metadata(byte_stream=input_file.open(), encoding=encoding)
+        with input_file.open() as input_stream:
+            return FromByteStream.table_metadata(byte_stream=input_stream, encoding=encoding)
```

Comment:
Do we support compressed streams? I think we should make sure we can read and write gzipped metadata files.